bb/bb.go

```diff
@@ -295,74 +299,97 @@ func oneFile(c Command, dir, s string, fset *token.FileSet, f *ast.File) error {
-			log.Fatalf("%v\n", err)
+		bbsetupPath := filepath.Join(config.Bbsh, "cmds", c.CmdName, "bbsetup.go")
+		if err := ioutil.WriteFile(bbsetupPath, b.Bytes(), 0444); err != nil {
+			return fmt.Errorf("error writing to %q: %v", bbsetupPath, err)
 		}
 		b.Reset()
+
 		// Write the file to interface to the command package.
 		t = template.Must(template.New("cmdFunc").Parse(cmdFunc))
 		if err := t.Execute(&b, c); err != nil {
-			log.Fatalf("spec %v: %v\n", cmdFunc, err)
+			return fmt.Errorf("spec %v: %v", cmdFunc, err)
 		}
-		fullCode, err := imports.Process("commandline", []byte(b.Bytes()), &opts)
+		fullCode, err := imports.Process("commandline", b.Bytes(), &opts)
 		if err != nil {
-			log.Fatalf("Main commandline imports: bad parse: '%v': %v", out, err)
+			return fmt.Errorf("main commandline imports: bad parse %q: %v", out, err)
 		}
-		if err := ioutil.WriteFile(filepath.Join(config.Bbsh, "cmd_"+c.CmdName+".go"), fullCode, 0444); err != nil {
-			log.Fatalf("%v\n", err)
+		cmdPath := filepath.Join(config.Bbsh, "cmd_"+c.CmdName+".go")
+		if err := ioutil.WriteFile(cmdPath, fullCode, 0444); err != nil {
+			return fmt.Errorf("error writing %q: %v", cmdPath, err)
 		}
 	}
 
 	return nil
 }
 
-func oneCmd(c Command) {
+func oneCmd(c Command) error {
 	// Create the directory for the package.
 	// For now, ./cmds/<package name>
 	packageDir := filepath.Join(config.Bbsh, "cmds", c.CmdName)
 	if err := os.MkdirAll(packageDir, 0755); err != nil {
-		log.Fatalf("Can't create target directory: %v", err)
+		return fmt.Errorf("failed to create target directory %q: %v", packageDir, err)
 	}
 
 	fset := token.NewFileSet()
 	c.FullPath = filepath.Join(c.Gopath, c.CmdPath)
 	p, err := parser.ParseDir(fset, c.FullPath, nil, parser.ParseComments)
 	if err != nil {
-		log.Printf("Can't Parsedir %v, %v", c.FullPath, err)
-		return
+		log.Printf("can't parsedir %q: %v", c.FullPath, err)
+		return nil
 	}
 
 	for _, f := range p {
 		for n, v := range f.Files {
-			oneFile(c, packageDir, n, fset, v)
+			if err := oneFile(c, packageDir, n, fset, v); err != nil {
+				return fmt.Errorf("error processing %s: %v", c.CmdName, err)
+			}
 		}
 	}
-	initMap += "\n\t\"" + c.CmdName + "\":" + c.CmdName + "Init,"
+
+	initMap += "\n\t\"" + c.CmdName + "\": " + c.CmdName + "Init,"
+
 	// In the bb case, the commands are built. In some cases, we want to
 	// specify init= to be a u-root command on boot. Hence, it now makes sense
 	// to have the ubin directory populated on boot, not by /init.
+	//
+	// Symlink /ubin/cmdname to /init.
 	l := filepath.Join(config.Bbsh, "ubin", c.CmdName)
 	if err := os.Symlink("/init", l); err != nil {
-		log.Fatalf("Symlinking %v -> /init: %v", l, err)
+		return fmt.Errorf("symlinking %v -> /init: %v", l, err)
 	}
+	return nil
 }
+
+func buildinit() error {
+	e := os.Environ()
+	e = append(e, "CGO_ENABLED=0")
+	e = append(e, "GO15VENDOREXPERIMENT=1")
```

Comment:
we don't really need this one VENDOREXPERIMENT thing any more?

Reply:
It's gone.